File: src/Controller.java
import javafx.scene.canvas.GraphicsContext;
import javafx.scene.paint.Color;
import javafx.scene.text.Text;

class Controller {
    static GraphicsContext gc;
    static Tetris game;
    static GraphicsContext gcNext;
    static Text score;
    static Text info;

    private static void drawRect(int i, int j, int figure, GraphicsContext gc, int offX, int offY) {
        if (figure > 0) {
            switch (figure) {
                case 1:
                    gc.setFill(Color.CYAN);
                    break;
                case 2:
                    gc.setFill(Color.ORANGE);
                    break;
                case 3:
                    gc.setFill(Color.BLUE);
                    break;
                case 4:
                    gc.setFill(Color.YELLOW);
                    break;
                case 5:
                    gc.setFill(Color.GREEN);
                    break;
                case 6:
                    gc.setFill(Color.PURPLE);
                    break;
                case 7:
                    gc.setFill(Color.RED);
                    break;
            }
            gc.fillRect(30 * j + 2 + offX, 30 * i + 2 + offY, 26, 26);
        }
    }

    static void refresh() {
        gc.clearRect(0, 0, 300, 600);
        for (int i = 0; i < 20; i++) {
            for (int j = 0; j < 10; j++) {
                drawRect(i, j, game.getCellValue(i, j), gc, 0, 0);
            }
        }
        gcNext.clearRect(0, 0, 120, 60);
        int offX;
        int offY;
        switch (game.getNextShapeNumber()) {
            case 1:
                offX = 0;
                offY = 15;
                break;
            case 4:
                offX = 30;
                offY = 0;
                break;
            default:
                offX = 15;
                offY = 0;
                break;
        }
        for (int i = 0; i < 2; i++) {
            for (int j = 0; j < 4; j++) {
                drawRect(i, j, game.getNextShapeCellValue(i, j), gcNext, offX, offY);
            }
        }
        score.setText(String.format("%03d", game.getScore()));

        if (game.isStopped()) {
            info.setFill(Color.valueOf("#aaaaaa"));
            info.setText("STOPPED");
        } else if (game.isGameOver()) {
            info.setFill(Color.RED);
            info.setText("GAME OVER");
        } else if (game.isPause()) {
            info.setFill(Color.valueOf("#aaaaaa"));
            info.setText("PAUSE");
        } else {
            info.setText("");
        }
    }
}

File: src/controller/Controller.kt
package controller

import game.Tetris
import javafx.scene.canvas.GraphicsContext
import javafx.scene.paint.Color
import javafx.scene.text.Text

object Controller {
    var gc: GraphicsContext? = null
    var game: Tetris? = null
    var gcNext: GraphicsContext? = null
    var score: Text? = null
    var info: Text? = null

    private fun drawRect(i: Int, j: Int, figure: Int, gc: GraphicsContext, offX: Int, offY: Int) {
        if (figure > 0) {
            when (figure) {
                1, 11 -> gc.fill = Color.CYAN
                2, 12 -> gc.fill = Color.rgb(255, 50, 0)
                3, 13 -> gc.fill = Color.BLUE
                4, 14 -> gc.fill = Color.YELLOW
                5, 15 -> gc.fill = Color.GREEN
                6, 16 -> gc.fill = Color.PURPLE
                7, 17 -> gc.fill = Color.RED
            }
            gc.fillRect((30 * j + 2 + offX).toDouble(), (30 * i + 2 + offY).toDouble(), 26.0, 26.0)
        }
    }

    fun refresh() {
        val tetris = game ?: return
        gc?.clearRect(0.0, 0.0, 300.0, 600.0)
        for (i in 2..21) {
            for (j in 0..9) {
                drawRect(i - 2, j, tetris.getCellValue(i, j), gc ?: return,
                        0, 0)
            }
        }
        gcNext?.clearRect(0.0, 0.0, 120.0, 60.0)
        val offX: Int
        val offY: Int
        when (tetris.nextShapeNumber) {
            1 -> {
                offX = 0
                offY = 15
            }
            4 -> {
                offX = 30
                offY = 0
            }
            else -> {
                offX = 15
                offY = 0
            }
        }
        for (i in 0..1) {
            for (j in 0..3) {
                drawRect(i, j, tetris.getNextShapeCellValue(i, j), gcNext ?: return,
                        offX, offY)
            }
        }
        score?.text = String.format("%03d", game?.score)

        when {
            tetris.isStopped -> {
                info?.fill = Color.valueOf("#aaaaaa")
                info?.text = "STOPPED"
            }
            tetris.isGameOver -> {
                info?.fill = Color.RED
                info?.text = "GAME OVER"
            }
            tetris.isPause -> {
                info?.fill = Color.valueOf("#aaaaaa")
                info?.text = "PAUSE"
            }
            else -> info?.text = ""
        }
    }
}
